handlers/simpleMath.go

```diff
@@ -0,0 +1,174 @@
+package handlers
+
+import (
+	"encoding/binary"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/vocdoni/blind-csp/types"
+	"go.vocdoni.io/dvote/db"
+	"go.vocdoni.io/dvote/db/metadb"
+	"go.vocdoni.io/dvote/log"
+)
+
+// SimpleMathHandler is a handler that requires a simple math operation to be resolved.
+type SimpleMathHandler struct {
+	kv         db.Database
+	keysLock   sync.RWMutex
+	mathRandom *rand.Rand
+}
+
+func (ih *SimpleMathHandler) addToken(token string, solution int) {
+	ih.keysLock.Lock()
+	defer ih.keysLock.Unlock()
+	tx := ih.kv.WriteTx()
+	defer tx.Discard()
+	n := make([]byte, 32)
+	binary.BigEndian.PutUint32(n, uint32(solution))
+	if err := tx.Set([]byte(token), n); err != nil {
+		log.Error(err)
+	}
+	if err := tx.Commit(); err != nil {
+		log.Error(err)
+	}
+}
+
+func (ih *SimpleMathHandler) getToken(token string) (int, error) {
+	ih.keysLock.RLock()
+	defer ih.keysLock.RUnlock()
+	tx := ih.kv.WriteTx()
+	defer tx.Discard()
+	v, err := tx.Get([]byte(token))
+	if err != nil {
+		return 0, err
+	}
+	return int(binary.BigEndian.Uint32(v)), nil
+}
+
+func (ih *SimpleMathHandler) delToken(token string) {
+	ih.keysLock.RLock()
+	defer ih.keysLock.RUnlock()
+	tx := ih.kv.WriteTx()
+	defer tx.Discard()
+	if err := tx.Delete([]byte(token)); err != nil {
+		log.Warn(err)
+	}
+	if err := tx.Commit(); err != nil {
+		log.Error(err)
+	}
+}
+
+// GetName returns the name of the handler
+func (ih *SimpleMathHandler) GetName() string {
+	return "simpleMath"
+}
+
+// Init initializes the handler.
+// Takes one argument for persistent data directory.
+func (ih *SimpleMathHandler) Init(opts ...string) (err error) {
+	ih.kv, err = metadb.New(db.TypePebble, filepath.Clean(opts[0]))
+	ih.mathRandom = rand.New(rand.NewSource(time.Now().UnixNano()))
+	return err
+}
+
+// Info returns the handler options and information.
+func (ih *SimpleMathHandler) Info() *types.Message {
+	return &types.Message{
+		Title:    "Simple math challenge",
+		AuthType: "blind",
+		AuthSteps: []*types.AuthField{
+			{Title: "Name", Type: "text"},
+			{Title: "Solution", Type: "int4"},
+		},
+	}
+}
+
+// Indexer takes a unique user identifier and returns the list of processIDs where
+// the user is elegible for participation. This is a helper function that might not
+// be implemented (depends on the handler use case).
+func (ih *SimpleMathHandler) Indexer(userID types.HexBytes) []*types.HexBytes {
+	userID[0] = 0xFF
+	userID[1] = 0xFF
```

Comment:
why would an indexer modify the input parameter?

Reply:
Its just a dummy example.